default/scripting/policies/COLONIALISM.focs.txt

```diff
@@ -44,14 +44,11 @@ Policy
                         Planet
                         OwnedBy empire = Source.Owner
                         Focus type = "FOCUS_INDUSTRY"
-                        ResourceSupplyConnected empire = Source.Owner condition = And [
-                            Not InSystem id = RootCandidate.SystemID
-                            Planet
-                            Species
-                            OwnedBy empire = Source.Owner
-                            Focus type = "FOCUS_INDUSTRY"
-                            WithinStarlaneJumps jumps = 3 condition = RootCandidate
-                        ]
+                        // TODO: this should actually be: a jump route not longer than 3 jumps where all hops are supplied
+                        //       currently one reaps benefits through blockades as long as both endpoints are in the same supply
+                        //       this would need a new condition able to exclude planets/ a filter based on the hops
+                        WithinStarlaneJumps jumps = 3 condition = RootCandidate
+                        ResourceSupplyConnected empire = Source.Owner condition = RootCandidate
```

Comment:
How it should work, I think:

for each effect target planet,
    sum "value" for all neutral populated planets that are within 3 jumps and supply connected to target, where
        value = neutral planet's industry / factor, where
            factor = count of target planets that will include the current neutral planet in their sums of "value"
    where target planets are owned by the empire, have industry focus, and are resource supply connected to a neutral populated planet within 1-3 jumps.

The count for each neutral planet should be the number of industry focused planets owned by the empire, with industry focus, that are supply connected to that neutral planet within 1-3 jumps. That is, those that will include that neutral planet in their "value" sum.

Reply:
yes, agreed. That was what i tried to implement modulus the current system exclusion, which should be removed following our discussion on the forums.